examples/ct/structures.go

```diff
@@ -0,0 +1,79 @@
+package ct
+
+// Code to handle encoding / decoding various data structures used in RFC 6962.
+
+import (
+	"crypto/sha256"
+	"errors"
+	"time"
+
+	"github.com/google/trillian/crypto"
+	"github.com/google/certificate-transparency/go/x509"
+	"github.com/google/certificate-transparency/go"
+	"github.com/google/trillian"
+)
+
+const millisPerNano int64 = 1000
+
+// GetCTKeyID takes the key manager for a log returns the LogID for it. (see RFC 6962 S3.2)
+// In CT V1 the log id is a hash of the public key.
+func GetCTLogID(km crypto.KeyManager) ([sha256.Size]byte, error) {
```

Comment:
The only place it's used so far is building a ct.SignedCertificateTimestamp, where it would have to be casted back. Not sure it's worth doing atm.